### ISSUE TITLE
Examples: amp-sidebar: autoscrolled element lacks programmatic state

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31174]
## WCAG Level
Level A

## Priority
High

## Type
Semantics

## Pages/screens/components affected
* [amp-sidebar](https://amp.dev/documentation/components/amp-sidebar/)

## Description
A control's role conveys to assistive technologies the purpose - and expected interaction or effect - of the control. Its state represents a particular characteristic of the component in response to an action or process.

Currently, in the `amp-sidebar` examples that show the effect of the `autoscroll` attribute (both in the documentation itself, and the separate [responsive sidebar with autofocus](https://daisy-purring-seal.glitch.me/sidebar-autoscroll.html) example), the element that is autoscrolled to presents a visual highlight/differentiation through CSS - but this is not programmatically conveyed to assistive technologies.

## User impact
When the state of a control is conveyed purely visually, this state will not be conveyed to users of assistive technologies.

In the case of the autoscrolling elements in the sidebar examples, these elements - despite their appearance - will not be announced any differently from the other elements in the sidebar.

## Required solution
Depending on what the visual highlighting represents/conveys, make sure that elements are denoted not just via CSS and styling, but that they have appropriate attributes that programmatically convey the same meaning.

In the two examples for `autoscroll`, the highlighted elements are intended to denote the navigation element referring to the current page.

```
<li autoscroll class="currentPage">Nav item 4</li>
```

```
<li autoscroll class=selected>Selected Nav Item</li>
```

In these cases, make sure to include the [`aria-current="page"` attribute](https://www.w3.org/TR/wai-aria-1.1/#aria-current).

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Turn on your screen reader.
2. Navigate through all controls on the page.
3. In the case of visually highlighted elements/controls, verify that the screen reader conveys the fact that the element/control is highlighted (this will depend on what the highlighting visually represents - e.g. if it conveys that a navigation control refers to the current page, then this fact should be conveyed).

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/TR/WCAG21/#name-role-value)

## More information
* [W3C - Understanding Success Criterion 4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
* [W3C - ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4.html)
* [W3C - WAI-ARIA Overview](https://www.w3.org/WAI/standards-guidelines/aria/)

## Test data
Test date: November 2020
